### PR TITLE
refactored createUrl #370

### PR DESCRIPTION
**Pull request by bother7**
instead of passing layer.name into the function, the layer is passed instead

### CODE REVIEW

**Comment by rlage** [comment]
Hi @bother7 , thank you for your PR! =). To improve you PR, could you please consider that the argument styleName also comes from the layer object?

I will add some comments in line to be clearer.

File: src/components/Api/GeoAPI.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ const GeoAPI = {
     layers: [],
     menu: [],
 
-    createUrl({layerName, styleName, clickData, featureCount}) {
-        return `?LAYERS=${layerName}&QUERY_LAYERS=${layerName}&STYLES=${styleName},&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetFeatureInfo&BBOX=${clickData.BBOX}&FEATURE_COUNT=${featureCount}&HEIGHT=${clickData.HEIGHT}&WIDTH=${clickData.WIDTH}&FORMAT=image%2Fpng&INFO_FORMAT=application%2Fjson&SRS=EPSG%3A4326&X=${clickData.X}&Y=${clickData.Y}&CQL_FILTER=1%3D1%3B1%3D1`
+    createUrl({layer, styleName, clickData, featureCount}) {
```
Comment (rlage) [inline src/components/Api/GeoAPI.js]: styleName is inside layer. You could check if it has style, you need to do something like this 
```
layer.styles[l.selectedLayerStyleId].name.replace('plataforma:', '')
```

File: src/components/LeafletMap/LeafletMapContainer.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ const mapDispatchToProps = (dispatch) => {
             // fetch layer data for clicked point if needed
             let urls = layers.map(l => {
                 return GeoAPI.createUrl({
-                    layerName: l.layerName,
+                    layer: l.layer,
```
Comment (rlage) [inline src/components/LeafletMap/LeafletMapContainer.js]: If you need to pass the object layer, which I believe is the case, you should pass only the " l ". `l.layer` here is undefined, I think.

**Comment by bother7** [comment]
i turned stylename into a blank string if it doesn't exist, is that alright or should it be null or undefined?

**Comment by nighto** [comment]
Hi @bother7! Your PR looks good. Can you please just update it removing `package-lock.json` and using 4-spaces indentation on `src/components/Api/GeoAPI.js`? Thanks!

**Comment by bother7** [comment]
Yup! sorry about that

**Comment by nighto** [comment]
Merged, thanks @bother7! Added your name to our [README](https://github.com/MinisterioPublicoRJ/inLoco-2.0/blob/develop/README.md) as a contributter.

**Pull request merged**